internal/service/user.go

```diff
@@ -35,10 +36,11 @@ func (service *User) CreateUser(ctx context.Context, signup request.Signup, pass
 	}
 
 	user := domain.User{
-		ID:       userid,
-		Email:    signup.Email,
-		Role:     signup.Role,
-		Password: string(encryptedPassword),
+		ID:                   userid,
+		Email:                signup.Email,
+		Role:                 signup.Role,
```

Comment:
I don't think we should accept "role" as a sign up parameter. 
Because in this case, basically anyone might register as an admin. We do not want that. This probably was glanced over during sign up review, since it was a lengthy one. We should have a default role (like "user") assigned upon signing up. Other roles should be assigned by admin directly.